src/foam/dao/index/LRUCachingIndexTest.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.index;
+
+import foam.nanos.auth.Country;
+import org.apache.commons.text.RandomStringGenerator;
+
+import java.io.IOException;
+
+import static org.apache.commons.text.CharacterPredicates.DIGITS;
+import static org.apache.commons.text.CharacterPredicates.LETTERS;
+
+public class LRUCachingIndexTest {
+
+  protected static final int SAMPLE_SIZE = 1000000;
+  protected static String[] samples = new String[SAMPLE_SIZE];
+  protected static RandomStringGenerator generator =
+      new RandomStringGenerator.Builder()
+          .withinRange('0', 'z')
+          .filteredBy(LETTERS, DIGITS)
+          .build();
+
+  public static void main(String[] args) throws IOException {
+    // generate random strings of length 50
+    for ( int i = 0; i < SAMPLE_SIZE; i++ ) {
+      samples[i] = generator.generate(50);
+    }
+
+    LRUCachingIndex index = new LRUCachingIndex(1000000,
+        new PersistedIndex("test", new TreeIndex(Country.CODE)));
+    long start = System.nanoTime();
+
+    for ( int i = 0; i < SAMPLE_SIZE; i++ ) {
+      index.wrap(samples[i]);
```

Comment:
Add some unwrap()'s also and then benchmark with and without the cache.

Reply:
I have a separate test which just sets the performance of the PersistedIndex. I'll add a benchmark for the unwraps too though.